app/src/behaviors/behavior_hold_tap.c

```diff
@@ -402,60 +404,72 @@ static inline const char *decision_moment_str(enum decision_moment decision_mome
 }
 
 static int press_hold_binding(struct active_hold_tap *hold_tap) {
+    struct zmk_behavior_binding binding = {.behavior_dev = hold_tap->config->hold_behavior_dev,
+                                           .param1 = hold_tap->param_hold};
     struct zmk_behavior_binding_event event = {
+        .binding = &binding,
         .position = hold_tap->position,
         .timestamp = hold_tap->timestamp,
+        .layer = hold_tap->layer,
+        .type = PRESS,
 #if IS_ENABLED(CONFIG_ZMK_SPLIT)
         .source = hold_tap->source,
 #endif
     };
 
-    struct zmk_behavior_binding binding = {.behavior_dev = hold_tap->config->hold_behavior_dev,
-                                           .param1 = hold_tap->param_hold};
-    return zmk_behavior_invoke_binding(&binding, event, true);
+    return raise_zmk_behavior_binding_event(event);
 }
 
 static int press_tap_binding(struct active_hold_tap *hold_tap) {
+    struct zmk_behavior_binding binding = {.behavior_dev = hold_tap->config->tap_behavior_dev,
```

Comment:
Here's where I'm concerned about the event pointing to a stack allocated binding.